### PR TITLE
[next] feat: fastify http server

### PR DESCRIPTION
**Pull request by pierissimo**
## Fastify support
This PR adds support for using Fastify as the underlying HTTP framework for DaVinci.
The HttpServer module provides a level of abstraction that allows switching from Express to Fastify (or the other way around) by changing just a few lines of code.

### Other changes included
- feat(App): added ability to listen for shutdown signals 
- chore(HttpServerModule): simplified HTTP server listen interface
- fix(HttpServerModule): trim slashes in urls

### CODE REVIEW

File: packages/core/src/App.ts [hunk 1]
```diff
@@ -3,33 +3,46 @@
...
-
 export interface AppOptions {
 	controllers?: ClassType[];
+	logger?: {
+		level?: Level | 'silent';
+	};
 }
 
 export class App extends Module {
+	logger = pino({ name: 'app' });
```
Comment (joseandrespg) [inline packages/core/src/App.ts:21]: Should we allow to customise the logger name?

File: packages/http-server-fastify/src/FastifyHttpServer.ts [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+
+type FastifyHttpServerModuleOptions = {
+	app?: FastifyInstance;
+	middlewares?: {
+		cors?: FastifyCorsOptions;
+	};
+	plugins?: [FastifyPluginCallback, FastifyPluginOptions?][];
+} & HttpServerModuleOptions;
+
+export class FastifyHttpServer extends HttpServerModule<
```
Comment (joseandrespg) [inline packages/http-server-fastify/src/FastifyHttpServer.ts:35]: Do we need to differentiate between what we extend from HttpServerModule and what we need to implement using `implements + Interface`?

File: packages/core/src/App.ts (hunk 1 above)
Comment (pierissimo) [reply packages/core/src/App.ts:21]: that'd be a good addition

File: packages/http-server-fastify/src/FastifyHttpServer.ts (hunk 2 above)
Comment (pierissimo) [reply packages/http-server-fastify/src/FastifyHttpServer.ts:35]: the specific express/fastify/other modules will have to extend the abstract HttpServerModule class, and that will:
- force them to implement the abstract method
- allow them to use some methods via `super`

What do you mean by differentiate them? Using some specific name convention?

Comment (joseandrespg) [reply packages/http-server-fastify/src/FastifyHttpServer.ts:35]: Just wondering if they will be able to not extend and do their own base class, but I'm fine with this if the intention is to make everyone extend this base class.

Comment (pierissimo) [reply packages/http-server-fastify/src/FastifyHttpServer.ts:35]: obviously they could, the HttpServerModule is meant just to provide the same interface for registering routes and bringing them with validation and interceptors functionalities

**Pull request merged**